app/src/main/java/org/love2d/android/LoveDocumentsProvider.java

```diff
@@ -0,0 +1,713 @@
+        Log.v(TAG, "moveDocument");
+        try {
+            // Copy document, insisting that the parent is correct
+            String newDocumentId = copyDocument(sourceDocumentId, sourceParentDocumentId,
+                    targetParentDocumentId);
+            // Remove old document
+            removeDocument(sourceDocumentId, sourceParentDocumentId);
+            return newDocumentId;
+        } catch (FileNotFoundException e) {
+            throw new FileNotFoundException("Failed to move document " + sourceDocumentId);
+        }
+    }
+    // END_INCLUDE(moveDocument)
+
+
+    @Override
+    public String getDocumentType(String documentId) throws FileNotFoundException {
+        File file = getFileForDocId(documentId);
+        return getTypeForFile(file);
+    }
+
+    /**
+     * @param projection the requested root column projection
+     * @return either the requested root column projection, or the default projection if the
+     * requested projection is null.
+     */
+    private static String[] resolveRootProjection(String[] projection) {
+        return projection != null ? projection : DEFAULT_ROOT_PROJECTION;
+    }
+
+    private static String[] resolveDocumentProjection(String[] projection) {
+        return projection != null ? projection : DEFAULT_DOCUMENT_PROJECTION;
+    }
+
+    /**
+     * Get a file's MIME type
+     *
+     * @param file the File object whose type we want
+     * @return the MIME type of the file
+     */
+    private static String getTypeForFile(File file) {
+        if (file.isDirectory()) {
+            return Document.MIME_TYPE_DIR;
+        } else {
+            return getTypeForName(file.getName());
+        }
+    }
+
+    /**
+     * Get the MIME data type of a document, given its filename.
+     *
+     * @param name the filename of the document
+     * @return the MIME data type of a document
+     */
+    private static String getTypeForName(String name) {
+        final int lastDot = name.lastIndexOf('.');
+        if (lastDot >= 0) {
+            final String extension = name.substring(lastDot + 1);
+            final String mime = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
+            if (mime != null) {
+                return mime;
+            }
+        }
+        return "application/octet-stream";
+    }
+
+    /**
+     * Gets a string of unique MIME data types a directory supports, separated by newlines.  This
+     * should not change.
+     *
+     * @param parent the File for the parent directory
+     * @return a string of the unique MIME data types the parent directory supports
+     */
+    private String getChildMimeTypes(File parent) {
+        Set<String> mimeTypes = new HashSet<String>();
+        mimeTypes.add("image/*");
+        mimeTypes.add("text/*");
+        mimeTypes.add("application/vnd.openxmlformats-officedocument.wordprocessingml.document");
```

Comment:
Is it possible to add `application/octet-stream`, `application/zip`, and `application/x-love-game` MIMEs? Or does `*/*` is possible and allowed?

Reply:
*/* should work